assets/src/components/with-term/index.js

```diff
@@ -33,30 +33,66 @@ export default function withTerm() {
 			const [ term, setTerm ] = useState( null );
 
 			useEffect( () => {
-				createTerm( taxonomy, label, value )
-					.catch( error => {
-						if ( error.code !== 'term_exists' ) {
-							return Promise.reject( error );
+				function handleSearchTerms( terms ) {
+					if ( terms.length ) {
+						setTerm( terms[ 0 ] );
+					}
+				}
+
+				function handleError( error ) {
+					if ( error.code !== 'term_exists' ) {
+						return Promise.reject( error );
+					}
+
+					return searchTerms( taxonomy, value ).then( searchTermsResponse => {
+						if ( searchTermsResponse instanceof Response ) {
+							searchTermsResponse.json().then(
+								terms => {
+									handleSearchTerms( terms );
+								}
+							);
+						} else {
+							handleSearchTerms( searchTermsResponse );
 						}
+					} );
+				}
+
+				function handeNewTerm( newTerm ) {
```

Comment:
Nitpick, it'd be nice to have a basic JSDoc block above these handler functions. Non-essential.